### DISCUSSION TITLE
New Dogecoin Address's vs New Transactions resources usage

### DISCUSSION THREAD

**Discussion by qlpqlp** [dogecoin/dogecoin#2761] [Help needed] [answered]
Hello to all :)
Hope u are all great :)

Questions for next "thing"

Does a wallet with xx Doge Address's each one with spendable Doge above 0, does it consume always resources even if on Paper/Offline Wallet? and if each address has no spendable Doge, 0? is it better?

Its better to have 1 Doge Address with minimum transactions or multiple Address's each one with minimum transactions?

Adding a new address, without any transactions pollutes resource usage?

What is the worst thing that can be done that can pollute the Dogecoin Blockchain?

Thanks in advance. Want to make sure to not make a "big mistake" on releasing something :)

**Comment by AbcSxyZ** [comment, marked as answer]
What does "create a wallet mean" ? It means choosing a random number between 0 (probably not starting from 0) and 2^160 (for btc), so you have 1461501637330902918203684832716283019655932542976 wallets possibilities.

How does the blockchain know about the random number you just picked ? The blockchain contain UTXO, unspend transanction output, the blockchain is a list of all transactions and everything is based on transactions. The miner create new coins, and then coins are just in circulation between people, exchanged.
The network will be aware of your new wallet when someone will send a new transaction to your wallet.

You can take a look at [Mastering Bitcoin](https://github.com/bitcoinbook/bitcoinbook), it can help you to understand more about what is going on behind the scene.

**Reply by qlpqlp** [reply]
Much thanks (yes i have stared to read last week lol, brain full of years of bad codding habits, need to empty a litle haha) 

Objective, big online store to opt to create one Doge Address to each client for payments or only one Dogecoin Address to all clientes to pay.

Want to make sure not to use Dogecoin Blockchain in the rong way that harms everyone on the future :)

**Reply by ReverseControl** [reply]
@qlpqlp We are striving to make doge handle massive amounts of transactions per block precisely for this use case: we already made the transactions 100x cheaper by default so that users/vendors like yourself do not have to think much about the transaction fees for these transaction heavy use cases. 

The number of transactions will be the same regardless of whether you make one address for all your store transactions or one for each transaction. One transaction per purchase. The real downside is for you as you will have to manage all those receiving wallets. From the Dogecoin Blockchain side it makes little difference.

At most, the worse cases is 2x the number of transactions for one-to-one wallets, where x = many-to-one wallet transactions. After the x transactions for all the purchases, you would do one more transaction for each wallet to consolidate all amounts into one wallet. So, it makes a difference, by a factor of 2, so, little difference.

On a simpler note,  making one wallet per transaction is better for privacy.
